pkg/geo/geo.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+// Package geo contains types that represent spatial objects on a plane or
+// sphere. These respectively correspond to the SQL GEOMETRY and GEOGRAPHY
+// types.
+//
+// Subpackages are available that perform operations using these types:
+// - geo/geofn implements the logic of the PostGIS/OGC functions over GEOMETRY
+//   and GEOGRAPHY types.
+// - geo/geoindex implements index-accelerated versions of a small number of the
+//   functions in geofn. These are used with a precomputed SQL inverted index.
+package geo
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/cockroachdb/errors"
+	"github.com/golang/geo/s2"
+)
+
+// Region represents a geometrical object in planar or spherical space.
+//
+// WIP This needs a better name. GeoJSON calls it a "Geometry", but that's
+// confusing with the SQL GEOMETRY type. Does PostGIS have a name for this? A
```

Comment:
postgis seems to call the data types `GEOMETRY` and `GEOGRAPHY`. they seem to both be serialized the same as a GSERIALIZED (http://postgis.net/docs/doxygen/2.5/df/d2d/structGSERIALIZED.html -- https://github.com/postgis/postgis/blob/master/liblwgeom/gserialized.txt), so maybe they can keep the same structure.

geography in returns `gserialized`: https://sourcegraph.com/github.com/postgis/postgis/-/blob/postgis/geography_inout.c#L136

geometry uses `LWGEOM_in` (https://sourcegraph.com/github.com/postgis/postgis/-/blob/postgis/postgis.sql.in#L113:24): https://sourcegraph.com/github.com/postgis/postgis/-/blob/postgis/lwgeom_inout.c#L84:1 which returns `GSERIALIZED` as well, so they are the same internally.

`Geography` seems to have a stricter set of types allowed in but that's about it.

geography can be explicitly cast to geometry, but geometry can be implicitly cast to geography